src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunction.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Drawing;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal class JsonFunctionImpl : JsonFunction, IAsyncTexlFunction4
+    {
+        public Task<FormulaValue> InvokeAsync(TimeZoneInfo timezoneInfo, IRContext irContext, FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            return Task.FromResult(new JsonProcessing(timezoneInfo, irContext, args).Process());
+        }
+
+        internal class JsonProcessing
+        {
+            private JsonFlags _flags;
```

Comment:
>JsonFlags [](http://example.com/codeflow?start=20&length=9)

can this be readonly

Reply:
This doesn't need to be a member of the class. It's only used in Process, and it can be a local variable in that method.

Reply:
@CarlosFigueiraMSFT Correct, I'll move it there